packages/slate/src/create-editor.ts

```diff
@@ -139,20 +142,31 @@ export const createEditor = (): Editor => {
       const { selection } = editor
 
       if (selection) {
-        return Node.fragment(editor, selection)
+        const fragment = Node.fragment(editor, selection)
+        return fragment
       }
+
       return []
     },
 
     insertBreak: () => {
       Transforms.splitNodes(editor, { always: true })
     },
 
-    insertFragment: (fragment: Node[]) => {
+    // @ts-ignore
+    insertFragment: (
+      fragment: Array<ElementOf<Editor<V>> | TextOf<Editor<V>>>
+    ) => {
       Transforms.insertFragment(editor, fragment)
     },
 
-    insertNode: (node: Node) => {
+    // @ts-ignore
+    insertNode: (
+      node:
+        | ElementOf<Editor<V>>
+        | TextOf<Editor<V>>
+        | Array<ElementOf<Editor<V>> | TextOf<Editor<V>>>
+    ) => {
```

Comment:
To tackle the circular ts error, we could set:

```suggestion
    insertNode: <EV extends V>(
      node:
        | VElement<V>
        | VText<V>
        | Array<VElement<V> | VText<V>>
    ) => {
```

where
```tsx
export type VElement<V extends Value> = V[number];
export type VText<V extends Value> = Extract<
  V[number]['children'][number],
  TText
>;
```
that way, we don't use `Editor<V>` but just `V`.

Note that `EV` generic is needed to avoid another ts error.